src/include/daos_errno.h

```diff
@@ -24,284 +24,212 @@ extern "C" {
+	/** Checksum error */                                                                      \
+	ACTION(DER_CSUM, (DER_ERR_DAOS_BASE + 21), Checksum error)                                 \
+	/** Unsupported durable format */                                                          \
+	ACTION(DER_DF_INVAL, (DER_ERR_DAOS_BASE + 22), Unsupported durable format)                 \
+	/** Incompatible durable format version */                                                 \
+	ACTION(DER_DF_INCOMPT, (DER_ERR_DAOS_BASE + 23), Incompatible durable format version)      \
+	/** Record size error */                                                                   \
+	ACTION(DER_REC_SIZE, (DER_ERR_DAOS_BASE + 24), Record size error)                          \
+	/** Used to indicate a transaction should restart */                                       \
+	ACTION(DER_TX_RESTART, (DER_ERR_DAOS_BASE + 25), Transaction should restart)               \
+	/** Data lost or not recoverable */                                                        \
+	ACTION(DER_DATA_LOSS, (DER_ERR_DAOS_BASE + 26), Data lost or not recoverable)              \
+	/** Operation canceled (non-crt) */                                                        \
+	ACTION(DER_OP_CANCELED, (DER_ERR_DAOS_BASE + 27), Operation canceled)                      \
+	/** TX is not committed, not sure whether committable or not */                            \
+	ACTION(DER_TX_BUSY, (DER_ERR_DAOS_BASE + 28), TX is not committed)                         \
+	/** Agent is incompatible with libdaos */                                                  \
+	ACTION(DER_AGENT_INCOMPAT, (DER_ERR_DAOS_BASE + 29), Agent is incompatible with libdaos)   \
+	/** Multiple shards locate on the same target */                                           \
+	ACTION(DER_SHARDS_OVERLAP, (DER_ERR_DAOS_BASE + 30), Shards overlap)                       \
+	/** #failures exceed RF(Redundancy Factor), data possibly lost */                          \
+	ACTION(DER_RF, (DER_ERR_DAOS_BASE + 31), Failures exceed RF)
 
 /** Defines the gurt error codes */
-#define D_FOREACH_ERR_RANGE(ACTION)	\
-	ACTION(GURT,	1000)		\
-	ACTION(DAOS,	2000)
+#define D_FOREACH_ERR_RANGE(ACTION)                                                                \
+	ACTION(GURT, 1000)                                                                         \
+	ACTION(DAOS, 2000)
 
-#define D_DEFINE_ERRNO(name, value, desc) name = value,
-#define D_DEFINE_ERRSTR(name, value, desc) #name,
+#define D_DEFINE_ERRNO(name, value, desc)   name = value,
+#define D_DEFINE_ERRSTR(name, value, desc)  #name,
 #define D_DEFINE_ERRDESC(name, value, desc) #desc,
 
-#define D_DEFINE_RANGE_ERRNO(name, base)			\
-	enum {							\
-		DER_ERR_##name##_BASE		=	(base),	\
-		D_FOREACH_##name##_ERR(D_DEFINE_ERRNO)		\
-		DER_ERR_##name##_LIMIT,				\
+#define D_DEFINE_RANGE_ERRNO(name, base)                                                           \
```

Comment:
(style)  Macros with multiple statements should be enclosed in a do - while loop
(style)  macros should not use a trailing semicolon